### PR TITLE
Adding GH action and updating the Makefile

### PR DESCRIPTION
**Pull request by betoredhat**
- Adding github action
- Updating target in Makefile
- Using golang image from mirror

### CODE REVIEW

**Review by krsacme — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+CLUSTER_CLI     ?= oc
+OPERATOR_NAME   := cnf-app-mac-operator
 
 # VERSION defines the project version for the bundle.
 # Update this value when you upgrade the version of your project.
 # To re-generate a bundle for another specific version without changing the standard setup, you can:
 # - use the VERSION as arg of the bundle target (e.g make bundle VERSION=0.0.2)
 # - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
-VERSION ?= 0.2.5
+
```
Comment (tonyskapunk) [inline Makefile:15]: Let's remove all these comments, I think `VERSION` is clear on what the purpose of it is.

File: Makefile [hunk 2]
```diff
@@ -112,6 +120,16 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
...
+# Build the operator image
+operator-build:
+	BUILDAH_FORMAT=docker ${CONTAINER_CLI} build . -t ${IMG}
+
+# Push the operator image
+operator-push:
+	${CONTAINER_CLI} push ${IMG}
+
+OS    = $(shell uname -s | tr '[:upper:]' '[:lower:]')
+ARCH  = $(shell uname -m | sed 's/x86_64/amd64/')
```
Comment (tonyskapunk) [inline Makefile:132]: Let's move these to the top with the rest of the var definition

**Review by tonyskapunk — CHANGES_REQUESTED** [review]
Few clean up changes and should be good to merge

File: Makefile (hunk 1 above)
Comment (betoredhat) [reply Makefile:15]: Fixed, thanks!

File: Makefile (hunk 2 above)
Comment (betoredhat) [reply Makefile:132]: Moded those, thanks

**Review by tonyskapunk — APPROVED** [review]
:+1:  Thanks!

**Pull request merged**